usr/share/rear/output/USB/Linux-i386/300_create_grub.sh

```diff
@@ -0,0 +1,70 @@
+# only run for grub
+# set nocasematch option
+shopt -s nocasematch
+if [ -z $USB_BOOTLOADER ] && [[ ! $USB_BOOTLOADER =~ grub ]]; then
+    return 0
+fi
+
+# we assume that REAL_USB_DEVICE and RAW_USB_DEVICE are both set from the script
+# in prep/USB/Linux-i386/350_check_usb_disk.sh
+
+[ "$RAW_USB_DEVICE" -a "$REAL_USB_DEVICE" ]
+BugIfError "RAW_USB_DEVICE and REAL_USB_DEVICE should be already set"
```

Comment:
Better avoid the ...IfError functions.
They do not work fail-safe, see the comment in
usr/share/rear/lib/_input-output-functions.sh
about "Using the ...IfError functions can result unexpected behaviour in certain cases"
currently at
https://github.com/rear/rear/blob/master/usr/share/rear/lib/_input-output-functions.sh#L773

We try to replace the ...IfError functions by plain bash code
where we find them when we change code
but we keep them as is in old code that we do not touch.

Reply:
I carried the check over from somewhere else. Will change it

Reply:
I can imagine wherefrom you copied it ;-)
cf. https://github.com/rear/rear/issues/2644#issuecomment-875406527
```
In particular the USB bootloader setup code is rather old code
e.g. 50% of the code in output/USB/Linux-i386/300_create_extlinux.sh
is 10 years old (from 2011 according to what 'git blame' shows)
so large parts of the USB bootloader setup code is not in compliance with
https://github.com/rear/rear/wiki/Coding-Style
```

Reply:
I was not 100% sure how to rewrite that so please double check my commit for that